File: code/sdk/android/cloudsheet-Android-sdk/src/main/kotlin/com/opencloudsheet/model/worksheet/IWorkSheet.kt
package com.opencloudsheet.model.worksheet

/**
 * Interface for worksheet operations with type-safe CRUD support.
 *
 * All data classes used with IWorkSheet must implement IWorksheetRow to enable
 * efficient update and delete operations via row index tracking.
 *
 * @param T The data class type that represents a row in the worksheet.
 *          Must implement IWorksheetRow.
 */
interface IWorkSheet<T : IWorksheetRow> {
    fun getId(): String
    fun getName(): String

    suspend fun get(): List<T>
    suspend fun create(row: T): T
    suspend fun update(row: T): T
    suspend fun delete(row: T): T
}

File: code/sdk/ios/OpenCloudSheet/Sources/OpenCloudSheet/Model/Worksheet/IWorkSheet.swift
//
//  IWorkSheet.swift
//  OpenCloudSheet
//
//  Interface for worksheet operations with type-safe CRUD support
//

import Foundation

public protocol IWorkSheet<T> {
    associatedtype T: IWorksheetRow

    func getId() -> String
    func getName() -> String

    func get() async throws -> [T]
    func create(row: T) async throws -> T
    func update(row: T) async throws -> T
    func delete(row: T) async throws -> T
}
